src/main/java/uk/org/tombolo/importer/DownloadUtils.java

```diff
@@ -42,13 +46,31 @@ public File fetchFile(URL url, String prefix, String suffix) throws IOException{
 	public InputStream fetchInputStream(URL url, String prefix, String suffix) throws IOException {
 		createCacheDir(prefix);
 		File localDatasourceFile = urlToLocalFile(url, prefix, suffix);
+
+
 		log.info("Fetching local file: {}", localDatasourceFile.getCanonicalPath());
 		if (!localDatasourceFile.exists()){
 			log.info("Local file not found: {} \nDownloading external resource: {}",
 												localDatasourceFile.getCanonicalPath(), url.toString());
-			URLConnection connection = url.openConnection();
-			if (suffix.equals(".json")) connection.setRequestProperty("Accept", "application/json");
-			return new TeeInputStream(connection.getInputStream(), new FileOutputStream(localDatasourceFile));
+
+			URLConnection urlConnection = url.openConnection();
+			if (suffix.equals(".json")) {
+				urlConnection.setRequestProperty("Accept", "application/json");
+			} else {
+				urlConnection.setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 " +
+								"(KHTML, like Gecko) Chrome/23.0.1271.95 Safari/537.11");
+			}
+			urlConnection.connect();
+
+			if (urlConnection instanceof HttpURLConnection) {
+				HttpURLConnection httpURLConnection = (HttpURLConnection) urlConnection;
+				int responseCode = httpURLConnection.getResponseCode();
+				if (responseCode != HttpURLConnection.HTTP_OK) {
+					log.warn(responseCode +": "+ httpURLConnection.getResponseMessage());
+				}
+			}
+
+			return new TeeInputStream(urlConnection.getInputStream(), new FileOutputStream(localDatasourceFile));
```

Comment:
even if the resposeCode is not 200 the method stills tries to return something, which i feel is not right, it should only try to return if the code is 200, else return may be null or something else, that would hint the calling class that something while downloading has gone wrong